app/services/force/soql/preference_service.rb

```diff
@@ -34,15 +34,31 @@ def buildAppPreferencesFilters(opts)
 
         filters = ''
         filters += "and Preference_All_Name__c like '%#{opts[:preference]}' " if opts[:preference].present?
-        filters += "and Application__r.Name like '%#{opts[:application_number]}%' " if opts[:application_number].present?
-        filters += "and Application__r.Applicant__r.First_Name__c like '%#{opts[:first_name]}%' " if opts[:first_name].present?
-        filters += "and Application__r.Applicant__r.Last_Name__c like '%#{opts[:last_name]}%' " if opts[:last_name].present?
         filters += "and Application__r.Processing_Status__c = " + (opts[:status] == 'No Status' ? 'NULL' : "'#{opts[:status]}'") if opts[:status].present?
         filters += "and Application__r.Has_ADA_Priorities_Selected__c INCLUDES (#{accessibility_string}) " if opts[:accessibility].present?
 
         filters
     end
 
+    def buildAppPreferencesSearch(search_terms_string)
+      # This takes a series of search terms and connects them all with 'OR's
+      # Across the provided fields
+      search_terms = search_terms_string.split(',')
+      search_fields = [
+        "Application__r.Name",
+        "Application__r.Applicant__r.First_Name__c",
+        "Application__r.Applicant__r.Last_Name__c",
+        "Application__r.Applicant__r.Email__c",
+      ]
+      search = []
+      for search_term in search_terms
+        for search_field in search_fields
+          search.push("#{search_field} like '%#{search_term}%'")
```

Comment:
do we have an idea of how much longer the query takes as the number of ORs increases?

Just wondering because if we want to eventually search all fields on the table we'll probably have to switch to sosl since this won't scale well, right?

Reply:
I looked into SOSL and it didn't really seem like it had the same ability to do where clauses and reference parents that SOQL does. 

I don't have a sense of how the performance scales except that I tried up to 26 different search phrases with the fields we have now and it didn't seem to impact performance.